### PR TITLE
Category navigation skill updates

### PR DESCRIPTION
**Pull request by JHurricane96**
- Remove rgb sensor and disable textures
- Add object segmentation sensor
- Enable kinematic mode
- Add new receptacle segmentation sensor
- Adjust robot radius
- Switch to discrete actions, fix discrete actions for hab 2.0
- OmegaConf speed fixes
- Spawn agent on active navmesh island
- Start episodes with retracted agent arm
- Disable auto sleep in kinematic mode

## Motivation and Context

Updates to get nav skill training working

## How Has This Been Tested

Launching training jobs and overfitting to a single episode

## Types of changes

- **\[Development\]** A pull request that add new features to the [habitat-lab](/habitat-lab) task and environment codebase. Development Pull Requests must be small (less that 500 lines of code change), have unit testing, very extensive documentation and examples. These are typically new tasks, environments, sensors, etc... The review process for these Pull Request is longer because these changes will be maintained by our core team of developers, so make sure your changes are easy to understand!

## Checklist

- [x] My code follows the code style of this project.
- [ ] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes if required.

### CODE REVIEW

File: habitat-lab/habitat/config/benchmark/rearrange/cat_nav_to_obj.yaml
Comment (yvsriram) [inline habitat-lab/habitat/config/benchmark/rearrange/cat_nav_to_obj.yaml:1]: You may want to repeat the changes in `cat_nav_to_rec.yaml` for the nav before Place

File: habitat-lab/habitat/tasks/rearrange/sub_tasks/cat_nav_to_obj_sensors.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        obs = np.copy(observations["robot_head_panoptic"])
+        obj_id_map = np.zeros(np.max(obs) + 1, dtype=np.int32)
+        for obj_id, semantic_id in task.receptacle_semantic_ids.items():
+            instance_id = obj_id + self._instance_ids_start
+            # Skip if receptacle is not in the agent's viewport
+            if instance_id >= obj_id_map.shape[0]:
+                continue
+            obj_id_map[instance_id] = semantic_id
+        obs = obj_id_map[obs]
+        # from habitat_sim.utils.viz_utils import observation_to_image
```
Comment (yvsriram) [inline habitat-lab/habitat/tasks/rearrange/sub_tasks/cat_nav_to_obj_sensors.py:61]: dead code

File: habitat-lab/habitat/config/habitat/task/rearrange/cat_nav_to_obj.yaml [hunk 2]
```diff
@@ -3,8 +3,12 @@
 defaults:
   - /habitat/task: task_config_base
   - /habitat/task/actions:
-    - base_velocity
-    - rearrange_stop
+    # - base_velocity
+    # - rearrange_stop
```
Comment (ykarmesh) [inline habitat-lab/habitat/config/habitat/task/rearrange/cat_nav_to_obj.yaml:7]: Lets either delete these or create another yaml file
```suggestion
```

File: habitat-lab/habitat/tasks/rearrange/sub_tasks/cat_nav_to_obj_sensors.py (hunk 1 above)
Comment (JHurricane96) [reply habitat-lab/habitat/tasks/rearrange/sub_tasks/cat_nav_to_obj_sensors.py]: Removed.

File: habitat-lab/habitat/config/habitat/task/rearrange/cat_nav_to_obj.yaml (hunk 2 above)
Comment (JHurricane96) [reply habitat-lab/habitat/config/habitat/task/rearrange/cat_nav_to_obj.yaml]: Removed.

File: habitat-lab/habitat/config/benchmark/rearrange/cat_nav_to_obj.yaml
Comment (JHurricane96) [reply habitat-lab/habitat/config/benchmark/rearrange/cat_nav_to_obj.yaml:1]: I think it's best we do that in another PR, this one's getting pretty big as it is.

File: habitat-lab/habitat/tasks/rearrange/sub_tasks/cat_nav_to_obj_task.py [hunk 3]
```diff
@@ -13,23 +16,52 @@
...
+    @property
+    def receptacle_semantic_ids(self):
+        return self._receptacle_semantic_ids
+
+    def reset(self, episode: Episode):
+        obs = super().reset(episode)
+        self._cache_receptacles()
+        return obs
+
+    def _cache_receptacles(self):
```
Comment (ASzot) [inline habitat-lab/habitat/tasks/rearrange/sub_tasks/cat_nav_to_obj_task.py:37]: In the future, this is probably good information to expose from the simulator. I think `habitat.datasets.rearrange.samplers.receptacle.find_receptacles` might also expose this same functionality?

**Review by ASzot — APPROVED** [review]
LGTM

File: habitat-lab/habitat/config/benchmark/rearrange/cat_nav_to_obj.yaml
Comment (yvsriram) [reply habitat-lab/habitat/config/benchmark/rearrange/cat_nav_to_obj.yaml:1]: Sure

**Review by yvsriram — APPROVED** [review]
LGTM

**Pull request merged**